README.txt
Copyright (C) 2008-2013 Free Software Foundation, Inc.
Copyright (C) 2009-2022 Vincent Goulet for the modifications.
See below for GNU Emacs license conditions.

Emacs Modified for macOS
========================

This is GNU Emacs 28.1 modified to include the following add-on
packages:

- ESS 18.10.3snapshot;
- AUCTeX 13.1;
- markdown-mode.el 2.5;
- exec-path-from-shell.el 1.12 to import the user's
  environment (by default PATH, MANPATH, LANG, TEXINPUTS and
  BIBINPUTS) at Emacs startup;
- dictionaries for the Hunspell spell checker (optional; see below for
  details);
- psvn.el r1573006 to work with Subversion repositories from
  within Emacs;
- default.el and site-start.el files to make everything work together.

The distribution is based on the latest stable release of GNU Emacs
compiled by David Caldwell (<https://emacsformacosx.com>).

In order to use Markdown you may need to install a parser such as
Pandoc (see <https://github.com/jgm/pandoc/releases/latest>) and
customize `markdown-command`.

You may want to customize `exec-path-from-shell-variables`.

Please direct questions or comments on this version of Emacs Modified
for macOS to Vincent Goulet <vincent.goulet@act.ulaval.ca>.

GNU Emacs Modified is free software: you can redistribute it and/or
modify it under the terms of the GNU General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.

Spell checking and dictionaries
===============================

Spell checking inside Emacs on macOS requires an external checker. I
recommend to install Hunspell (<https://hunspell.github.io>) using
Homebrew (<https://brew.sh>).

The Hunspell installation does not include any dictionaries.
Therefore, this distributions of Emacs ships with the following Libre
Office dictionaries suitable for use with Hunspell:

- English (version 2022.07.01);
- French (version 5.7);
- German (version 2017.01.12);
- Spanish (version 2.5).

Copy the files in the `Dictionaries` directory of the disk image to
`~/Library/Spelling`. If needed, create a symbolic link named after
your LANG environment variable to the corresponding dictionary and
affix files. For example, if LANG is set to fr_CA.UTF-8, do from the
command line

  cd ~/Library/Spelling
  ln -s fr-classique.dic fr_CA.dic
  ln -s fr-classique.aff fr_CA.aff

Finally, if you have a Mac with an Apple Silicon CPU (M1 and above),
add the following lines to your ~/.emacs file:

  (setq-default ispell-program-name "/opt/homebrew/bin/hunspell")
  (setq ispell-really-hunspell t)

For an Intel CPU, use instead:

  (setq-default ispell-program-name "/usr/local/bin/hunspell")
  (setq ispell-really-hunspell t)

Spell checking should now work with `M-x ispell`.

See <https://extensions.libreoffice.org/extensions> to install
additional dictionnaries.

GNU Emacs
=========

[The following are excerpts from the file etc/NEXTSTEP in the GNU
Emacs sources.]

The Nextstep support code works on many POSIX systems (and possibly
W32) using the GNUstep libraries, and on MacOS X systems using the
Cocoa libraries.

GNU Emacs is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

GNU Emacs is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.


NEWS
===
=== Emacs Modified for macOS
===

# VERSION 28.1-4-modified-1 (26 July 2022)

## Updates

- Emacs 28.1 released 4 April 2022 based off the macOS binary release
  28.1-4 of David Caldwell.
- ESS 18.10.3snapshot, actually the tip of the development branch as
  of this distribution's release date.
- AUCTeX 13.1
- markdownmode.el v2.5
- Libre Office English dictionary 2022.07.01

## Changes

- Org is removed from the distribution as version 9.5 is included in
  Emacs 28. Installation from the sources is deprecated anyway.
- David Holm's TabBarMode is removed from the distribution as it does
  not compile with Emacs 28.1.
- Since a directory 'site-lisp' inside the application bundle (e.g.
  '/Applications/Emacs.app/Contents/Resources/site-lisp/') is no
  longer included in 'load-path' by default, the extensions are
  installed in the 'lisp' directory instead.


# VERSION 27.2-2-modified-1 (28 March 2021)

## Updates

- Emacs 27.2-2 released 25 March 2021
- AUCTeX 12.3
- Org 9.4
- Libre Office English dictionary 2021.02.01


# VERSION 27.1-1-modified-1 (16 September 2020)

## Updates

- Emacs 27.1-1 released 10 August 2020
- Org 9.3.8
- markdownmode.el v2.4
- Libre Office English dictionary 2020.08.01
- Libre Office Spanish dictionary 2.5

## Changes

- Largely obsolete framepop.el removed from the distribution.


# VERSION 26.3-modified-2 (11 March 2020)

## New feature

The distribution now ships with dictionaries to use with Hunspell. You
need to install an Hunspell binary separately, most likely through
Homebrew. You may then use the included English, French, German and
Spanish dictionaries by copying the files in the `Dictionaries`
directory to `~/Library/Spelling`. See the README.txt file for
additional details.

## Updates

- AUCTeX 12.2
- Org 9.3.6


# VERSION 26.3-modified-1 (27 September 2019)

## Updates

- Emacs 26.3 released 28 August 2019
- Org 9.2.6


# VERSION 26.2-modified-2 (5 July 2019)

## New feature

David Holm's TabBarMode (v2.2), a minor mode to display a tab bar at
the top of the Emacs window, similar to the idea of web browsers tabs.
Brought to you by Rodney Sparapani <rsparapa@mcw.edu>.

Note that TabBarMode is *not* enabled by default. To use it, use `M-x
tabbar-mode` or add `(tabbar-mode)` in your `~/.emacs` file.

## Updates

- Org 9.2.4
- exec-path-from-shell 1.12

## User visible changes

In order to have a more complete AUCTeX and RefTeX setup, the
environment variables TEXINPUTS and BIBINPUTS are now also imported
from the shell at startup. Therefore, the following environment
variables are imported at startup by default: PATH, MANPATH, LANG,
TEXINPUTS, BIBINPUTS.


# VERSION 26.2-modified-1 (19 April 2019)

## Updates

- Emacs 26.2 released 14 April 2019
- Org 9.2.3


# VERSION 26.1-2-modified-2 (11 November 2018)

## Updates

- ESS 18.10.2 (patched to include the *.el files in the distribution)
- Org 9.1.14

## User visible changes

- Polymode is removed from the distribution. See
  https://polymode.github.io/ to install the package.


# VERSION 26.1-2-modified-1 (31 May 2018)

## Updates

- Emacs 26.1 released 28 May 2018
- Org 9.1.13

# VERSION 25.3-modified-2 (18 January 2018)

## Updates

- AUCTeX 12.1
- Org 9.1.6


# VERSION 25.3-modified-1 (16 November 2017)

This version brings many package updates. I realize I somewhat fell
behind!

- Emacs 25.3
- ESS 17.11
- AUCTeX 11.91
- Org 9.1.3
- markdownmode.el v2.3


# VERSION 25.2-modified-2 (2 June 2017)

## Updates

- Org 9.0.7
- polymode 2017-03-07
- markdownmode.el v2.2

## Bug fix

- markdownmode.el was broken in the previous release.


# VERSION 25.2-modified-1 (4 May 2017)

## Updates

- GNU Emacs 25.2, released April 21, 2017.
- Org 9.0.6

## User visible changes

- In `site-start.el`, `ess-nuke-trailing-whitespace` is now added to
  the `write-contents-functions` hook to avoid bleeding of the
  functionality to other major modes (most notably markdown, where
  trailing whitespace is sometimes useful).


# VERSION 25.1-1-modified-5 (11 January 2016)

## Updates

- AUCTeX 11.90


# VERSION 25.1-1-modified-4 (9 January 2016)

*This is essentially a release to setup the new release mechanism through GitHub*

## Updates

- Org 9.0.3


# VERSION 25.1-1-modified-3 (15 December 2016)

## Updates

- GNU Emacs 25.1-1, released September 20, 2016 by David Caldwell.
- Org 9.0.2

## Bug fix

- `emacs-modified-version` now correctly report the Emacs patch level.


# VERSION 25.1-modified-2 (24 October 2016)

## Updates

- ESS 16.10


# VERSION 25.1-modified-1 (19 September 2016)

## Updates

- GNU Emacs 25.1, released September 17, 2016.
- Org 8.3.6


# VERSION 24.5-1-modified-7 (15 July 2016)

This is a re-release fixing only the current and the README.txt
files. The 8 May 2016 version of this file should have also included
the following statement.

## User visible changes

- The default configuration has changed as follows:
  ess-swv-plug-into-AUCTeX is active for Sweave documents using the
  AUCTeX interface; polymode is active only for Rmarkdown (and as a
  by-product for R help files). See threads in the ess-help mailing
  list in April and July 2016 for some rationale for this decision
  (https://stat.ethz.ch/pipermail/ess-help/2016-April/010932.html).

# VERSION 24.5-1-modified-7 (8 May 2016)

## Updates

- ESS 16.04


# VERSION 24.5-1-modified-6 (30 March 2016)

## New feature

- markdown-mode is now automatically loaded for file extensions
  `.text`, `.markdown` and `.md`.

## Updates

- AUCTeX 11.89
- Org 8.3.4
- markdown-mode 2.1
- polymode 2016-03-24
- exec-path-from-shell 1.10-7


# VERSION 24.5-1-modified-5 (15 December 2015)

## Bug fix

- The patches to psvn.el to support Subversion > v1.7, removed from
  the previous release, are reinstated.


# VERSION 24.5-1-modified-4 (11 December 2015)

## Updates

- ESS 15.09-2


# VERSION 24.5-1-modified-3 (26 October 2015)

## Updates

- ESS 15.09-1
- Org 8.3.2

## (Possibly) User visible change


default.el
;;; default.el --- Default configuration for Emacs Modified for macOS
;;
;; This file is used mainly to load custom extensions. It is loaded
;; *after* any user and site configuration files.
;;
;; Copyright (C) 2009-2022 Vincent Goulet
;;
;; Author: Vincent Goulet
;;
;; This file is part of Emacs Modified for macOS
;; https://gitlab.com/vigou3/emacs-modified-macos

;; Emacs Modified for macOS is free software; you can redistribute it
;; and/or modify it under the terms of the GNU General Public License
;; as published by the Free Software Foundation; either version 3, or
;; (at your option) any later version.
;;
;; GNU Emacs is distributed in the hope that it will be useful,
;; but WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;; GNU General Public License for more details.
;;
;; You should have received a copy of the GNU General Public License
;; along with GNU Emacs; see the file COPYING.  If not, write to the
;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
;; Boston, MA 02110-1301, USA.

;;;
;;; Version number of Emacs Modified for macOS
;;;
;; Define variable and function 'emacs-modified-version'
(require 'version-modified)

;;;
;;; Import the shell environment
;;;
;; Import some shell environment variables into Emacs at launch. Steve
;; Purcell's exec-path-from-shell imports PATH and MANPATH by default;
;; LANG, TEXINPUTS and BIBINPUTS are added here. You can customize
;; 'exec-env-from-shell-variables' in site-start.el or the user's
;; config file.
(require 'exec-path-from-shell)
(nconc exec-path-from-shell-variables '("LANG" "TEXINPUTS" "BIBINPUTS"))
(exec-path-from-shell-initialize)


site-start.el
;;; site-start.el --- Customizations for Emacs Modified for macOS
;;
;; Copyright (C) 2009-2022 Vincent Goulet
;;
;; Author: Vincent Goulet
;;
;; This file is part of Emacs Modified for macOS
;; https://gitlab.com/vigou3/emacs-modified--macos

;; Emacs Modified for macOS is free software; you can redistribute it
;; and/or modify it under the terms of the GNU General Public License
;; as published by the Free Software Foundation; either version 3, or
;; (at your option) any later version.
;;
;; GNU Emacs is distributed in the hope that it will be useful,
;; but WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;; GNU General Public License for more details.
;;
;; You should have received a copy of the GNU General Public License
;; along with GNU Emacs; see the file COPYING.  If not, write to the
;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
;; Boston, MA 02110-1301, USA.


;;;
;;; Environment setting
;;;
;; By default, environment variables PATH, MANPATH, LANG, TEXINPUTS
;; and BIBINPUTS are imported from the shell at startup. If needed,
;; more variables may be imported by listing them in
;; 'exec-path-from-shell-variables' variable. You may also Customize
;; this variable.
;;
;; Customizations should probably go in a user's .emacs file, so just
;; consider the code below as an example of how to do it. Bear in mind
;; that LANG, TEXINPUTS and BIBINPUTS are added to
;; 'exec-path-from-shell-variables' in default.el, so there is no need
;; to include them in the definition.
;(setq exec-path-from-shell-variables '("PATH" "MANPATH" "FOO" "BAR"))

;;;
;;; Nice options to have On by default
;;;
(mouse-wheel-mode t)			; activate mouse scrolling
(global-font-lock-mode t)		; syntax highlighting
(transient-mark-mode t)			; sane select (mark) mode
(delete-selection-mode t)		; entry deletes marked text
(show-paren-mode t)			; match parentheses
(add-hook 'text-mode-hook 'turn-on-auto-fill) ; wrap long lines in text mode

;;;
;;; ESS
;;;

;; Add the path to ESS to 'load-path'.
(add-to-list 'load-path 
	     (let ((default-directory
		     (file-name-directory load-file-name)))
	       (expand-file-name "ess/lisp")))

;; Following the "source is real" philosophy put forward by ESS, one
;; should not save the workspace at the end of an R session. Hence,
;; the option is disabled here.
(setq-default inferior-R-args "--no-save ")

;; Enable sweaving directly within the AUCTeX ecosystem.
(setq-default ess-swv-plug-into-AUCTeX-p t)

;; Automagically delete trailing whitespace when saving R script
;; files. One can add other commands in the ess-mode-hook below.
(add-hook 'ess-mode-hook
	  '(lambda()
	     (add-hook 'write-contents-functions
		       (lambda ()
                         (ess-nuke-trailing-whitespace)))
	     (setq ess-nuke-trailing-whitespace-p t)))

;; Load ESS when needed.
(load "ess-autoloads")

;;;
;;; AUCTeX
;;;
;; Turn on RefTeX for LaTeX documents. Put further RefTeX
;; customizations in your .emacs file.
(add-hook 'LaTeX-mode-hook
	  (lambda ()
	    (turn-on-reftex)
	    (setq reftex-plug-into-AUCTeX t)))

;; Defensive hack to find latex in case the PATH environment variable
;; was not correctly altered at TeX Live installation. Contributed by
;; Rodney Sparapani <rsparapa@mcw.edu>.
(require 'executable)
(if (and (not (executable-find "latex")) (file-exists-p "/usr/texbin"))
    (setq LaTeX-command-style
	  '(("" "/usr/texbin/%(PDF)%(latex) %S%(PDFout)"))))

;; Load AUCTeX and preview-latex.
(load "auctex.el" nil t t)
(load "preview-latex.el" nil t t)

;;;
;;; markdown-mode
;;;
;; Activation of markdown-mode for common file extensions. One will
;; also need to install a Markdown parser (such as 'pandoc') and then
;; to indicate its location by customizing the variable
;; 'markdown-command'.
(add-to-list 'auto-mode-alist '("\\.text\\'" . markdown-mode))
(add-to-list 'auto-mode-alist '("\\.markdown\\'" . markdown-mode))
(add-to-list 'auto-mode-alist '("\\.md\\'" . markdown-mode))
(autoload 'markdown-mode "markdown-mode"
   "Major mode for editing Markdown files" t)

;;;
;;; SVN
;;;
;; Support for the Subversion version control system. Use 'M-x
;; svn-status RET' on a directory under version control to update,
;; commit changes, revert files, etc., all within Emacs.
(add-to-list 'vc-handled-backends 'SVN)
(add-hook 'svn-log-edit-mode-hook 'turn-off-auto-fill) ; useful option
(require 'psvn)
